packages/core/src/utils/customStyles/objectPosition.ts

```diff
@@ -0,0 +1,33 @@
+import { style } from 'styled-system'
+
+export const objectPositionValues = [
+  'left',
+  'right',
+  'top',
+  'topLeft',
+  'topRight',
+  'bottom',
+  'bottomLeft',
+  'bottomRight',
+] as const
+
+export type ObjectPosition = typeof objectPositionValues[number]
+
+const OBJECT_POISITION_PROPS_TO_CSS_VALUE = {
+  left: 'left',
+  right: 'right',
+  top: 'top',
+  topLeft: 'top left',
+  topRight: 'top right',
+  bottom: 'bottom',
+  bottomLeft: 'bottom left',
+  bottomRight: 'bottom right',
+}
+
+export function objectPosition() {
+  return style({
+    prop: 'objectPosition',
+    cssProperty: 'object-position',
+    transformValue: (val) => OBJECT_POISITION_PROPS_TO_CSS_VALUE[val] || 'center',
```

Comment:
`style` handles if `val` is an `Array` ([source](https://github.com/styled-system/styled-system/blob/v4.2.2/src/index.js#L87))